tests/pact/lib.repl

```diff
@@ -3,13 +3,6 @@
 
 (expect "env-hash" "Set tx hash to YQo" (env-hash "YQo"))
 
-;; versions of base64-bytestring < 1.2 accept invalid base64 encodings. The
-;; following test case is expected to fail after upgrading to
-;; base64-bytestring > 1.2.0.0. This would be a breaking change and it must
-;; be guaranteed that mainnet history isn't affected.
-;; (expect-failure "env-hash invalid base64" (env-hash "aa"))
-(expect "env-hash invalid base64" "Set tx hash to aa" (env-hash "aa"))
```

Comment:
why are we able to get rid of this test?

Reply:
IIUC the test was mostly here to ensure that we don't accidentally bump base64-bytestring. It did this by asserting some incorrect behavior that's specific to the legacy base64-bytestring version.

We are bumping base64-bytestring on purpose, and we're replaced this deleted test with other tests that ensure old behavior doesn't change (prior to the fork).

We don't want this particular test anymore because we don't need to support users setting the env-hash to some invalid value. (`env-hash` itself isn't relevant to on-chain compatibility). Does that all sound right?

Reply:
I'm not really the person to know, @emilypi or @jmcardon should opine.

Reply:
So the good news is that we never enforce that the env-hash set with env-hash is ever anything other than parseable text.

```haskell
envHash :: RNativeFun LibState
envHash i [TLitString s] = case fromText' s of
  Left err -> evalError' i $ "Bad hash value: " <> pretty s <> ": " <> prettyString err
  Right h -> do
    setenv eeHash h
    return $ tStr $ "Set tx hash to " <> s
envHash i as = argsError i as
```

It just so happens that all of our examples show something like `(env-hash (hash "foo"))` which automatically generates a base64url-nopad text value in the hash call. However, for other inputs like raw strings, we just make sure it's a string and go "yep that's a string" and set it without any more thought. When it comes to actual prod hashes, those tx hashes are generated through the apireq json encoding for hashes which does the right thing in the ToJSON instance for hashes, so it's not touched by anything @imalsogreg did in his PR. We can rely on those to be well-structured because they're b64 encode, which is not an issue for us. Only the b64 decode was a problem, and only with "untrusted" b64 text (untrusted in the sense that we didn't produce it).

This would be a fun 15 minute PR to lock down what we want out of env-hash but it's orthogonal to webauthn

Reply:
Investigated with @imalsogreg and this will be an issue for the `env-hash` native exclusively, which does not hit the tx hashes for prod cmds. Submitting a follow up to handle it.